install.sh

```diff
@@ -143,48 +139,28 @@ if [ -d "${CREW_PREFIX}" ]; then
     echo_info "Continue?"
     select continue in "Yes" "No"; do
       if [[ "${continue}" == "Yes" ]]; then
-        # Let's see if avoiding sudo breaks anything.
-        # sudo find "${CREW_PREFIX}" -mindepth 1 -delete
-        find "${CREW_PREFIX}" -mindepth 1 -delete
+        sudo find "${CREW_PREFIX}" -mindepth 1 -delete
         break 2
       else
         exit 1
       fi
     done
   fi
 else
-  echo_error "${CREW_PREFIX} does not exist. (It should be created or mounted prior to install!)"
-  exit 1
+  mkdir -p "${CREW_PREFIX}" || sudo mkdir -p "${CREW_PREFIX}"
 fi
 
-# This will allow things to work without sudo.
-# sudo chown "$(id -u)":"$(id -g)" "${CREW_PREFIX}"
-
-perm_error () {
-  if tty | grep -q /dev/pts1; then
-    echo_success "You are in VT-2."
-    if [ "${EUID}" == "0" ]; then
-      echo_info "Please run: 'chown -R chronos ${CREW_PREFIX}', and then login as 'chronos' and restart the install."
-      echo_info "Note that you can set the chronos password if you login as root and run 'chromeos-setdevpasswd'."
-      exit 1
-    fi
-  fi
-  # Give message on how to fix permissions in VT-2.
-  echo_error "You need set permissions in the VT-2 shell for the install to work."
-  echo_info "Please use Ctrl-Alt-{F2/Right arrow/Refresh}) to switch to VT-2."
-  echo_info "Then login as 'root' and run: 'chown -R chronos ${CREW_PREFIX}'"
-  echo_info "or login as 'chronos' and run: 'sudo chown -R chronos ${CREW_PREFIX}'"
-  echo_info "Then switch back to crosh in the main ChromeOS screen with Ctrl-Alt-{F1/Left arrow}) and restart the install."
-  exit 1
-}
-
-
+# Do not redundantly use sudo if the user already owns the directory.
+if [ "$(stat -c '%u' "${CREW_PREFIX}")" == "$(id -u)" ]; then
+  # This will allow things to work without sudo.
+  sudo chown "$(id -u)":"$(id -g)" "${CREW_PREFIX}"
+fi
 
-# This will create the directories
+# This will create the directories.
 crew_folders="bin cache doc docbook include lib/crew/packages lib$LIB_SUFFIX libexec man sbin share var etc/crew/meta etc/env.d tmp/crew/dest"
 # shellcheck disable=SC2086
 # Quoting crew_folders leads to breakage
-(cd "${CREW_PREFIX}"; mkdir -p ${crew_folders} || perm_error )
+(cd "${CREW_PREFIX}"; mkdir -p ${crew_folders})
```

Comment:
`&&` would be better than `;` in case the cd command fails and we create crew folders in some random place.